Property 'ticks' does not exist on type 'Axis' in Angular 7

Update
While the app will run in a development environment with this error, it causes the build to fail.  
Update 2
Found the @ts-ignore directive so can get through the build


#### Expected behaviour
Compile without errors when accessing yAxis[0].ticks in chart load event.  

#### Actual behaviour
Get error message: Property 'ticks' does not exist on type 'Axis'

#### Live demo with steps to reproduce
The error can be reproduced by:
1) install the highcharts-angular project from github
2) Using VS Code, add the following code into hcOptions in app.component at line 121: 
```
	chart: {
		events: {
			load: function () {
				this.yAxis[0].ticks[6].label.translate(-50, 0);
				}
			}
		},
```
3) run 'npm start'
4) System displays "ERROR in src/app/app.component.ts(125,19): error TS2339: Property 'ticks' does not exist on type 'Axis'."

Created the stackblitz below ... but stackblitz does _not_ display typescript compile errors (that I can find).
https://stackblitz.com/edit/angular-gauge-label-positioning?file=src/app/hello.component.ts


#### Product version
Latest version of highcharts-angular as of 9/11/19:
 - highcharts 7.1.1
 - highcharts-custom-events 2.2.2
 - node 8.12.0
 - npm 6.4.1
 - VS Code 1.38
 - Angular 7.1.0

#### Affected browser(s)
Chrome

Note: 
The code does run.  Just trying to eliminate the compile error.

--- a/js/parts/Axis.js
+++ b/js/parts/Axis.js
@@ -3112,7 +3112,6 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
          *
          * @see {@link Highcharts.Tick}
          *
-         * @private
          * @name Highcharts.Axis#ticks
          * @type {Highcharts.Dictionary<Highcharts.Tick>}
          */
@@ -3123,7 +3122,6 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
          *
          * @see {@link Highcharts.Tick}
          *
-         * @private
          * @name Highcharts.Axis#minorTicks
          * @type {Highcharts.Dictionary<Highcharts.Tick>}
          */
@@ -3229,9 +3227,14 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
      * The default label formatter. The context is a special config object for
      * the label. In apps, use the
      * [labels.formatter](https://api.highcharts.com/highcharts/xAxis.labels.formatter)
-     * instead except when a modification is needed.
-     * @private
+     * instead, except when a modification is needed.
+     *
+     * @function Highcharts.Axis#defaultLabelFormatter
+     *
+     * @this Highcharts.AxisLabelsFormatterContextObject
+     *
      * @return {string}
+     * The formatted label content.
      */
     defaultLabelFormatter: function () {
         var axis = this.axis, value = this.value, time = axis.chart.time, categories = axis.categories, dateTimeLabelFormat = this.dateTimeLabelFormat, lang = defaultOptions.lang, numericSymbols = lang.numericSymbols, numSymMagnitude = lang.numericSymbolMagnitude || 1000, i = numericSymbols && numericSymbols.length, multi, ret, formatOption = axis.options.labels.format, 

--- a/js/parts/Tick.js
+++ b/js/parts/Tick.js
@@ -39,13 +39,13 @@ var correctFloat = H.correctFloat, destroyObjectProperties = H.destroyObjectProp
  * The axis of the tick.
  *
  * @param {number} pos
- * The position of the tick on the axis.
+ * The position of the tick on the axis in terms of axis values.
  *
  * @param {string} [type]
- * The type of tick.
+ * The type of tick, either 'minor' or an empty string
  *
  * @param {boolean} [noLabel=false]
- * Wether to disable the label or not. Defaults to false.
+ * Whether to disable the label or not. Defaults to false.
  *
  * @param {object} [parameters]
  * Optional parameters for the tick.
@@ -58,7 +58,7 @@ H.Tick = function (axis, pos, type, noLabel, parameters) {
      */
     this.axis = axis;
     /**
-     * The logical position of the tick on the axis.
+     * The logical position of the tick on the axis in terms of axis values.
      * @name Highcharts.Tick#pos
      * @type {number}
      */
@@ -107,14 +107,16 @@ H.Tick.prototype = {
         }
         // set properties for access in render method
         /**
-         * True, if the tick is the first one on the axis.
+         * True if the tick is the first one on the axis.
          * @name Highcharts.Tick#isFirst
+         * @readonly
          * @type {boolean|undefined}
          */
         tick.isFirst = isFirst;
         /**
-         * True, if the tick is the last one on the axis.
+         * True if the tick is the last one on the axis.
          * @name Highcharts.Tick#isLast
+         * @readonly
          * @type {boolean|undefined}
          */
         tick.isLast = isLast;
@@ -269,7 +271,7 @@ H.Tick.prototype = {
         }
     },
     /**
-     * Gets the x and y positions for ticks.
+     * Gets the x and y positions for ticks in terms of pixels.
      *
      * @private
      * @function Highcharts.Tick#getPosition

--- a/js/parts/Tick.js
+++ b/js/parts/Tick.js
@@ -32,32 +32,52 @@ var correctFloat = H.correctFloat, destroyObjectProperties = H.destroyObjectProp
 /**
  * The Tick class.
  *
- * @private
  * @class
  * @name Highcharts.Tick
  *
  * @param {Highcharts.Axis} axis
+ * The axis of the tick.
  *
  * @param {number} pos
- *        The position of the tick on the axis.
+ * The position of the tick on the axis.
  *
  * @param {string} [type]
- *        The type of tick.
+ * The type of tick.
  *
  * @param {boolean} [noLabel=false]
- *        Wether to disable the label or not. Defaults to false.
+ * Wether to disable the label or not. Defaults to false.
  *
- * @param {Highcharts.TickParametersObject} [parameters]
- *        Optional parameters for the tick.
+ * @param {object} [parameters]
+ * Optional parameters for the tick.
  */
 H.Tick = function (axis, pos, type, noLabel, parameters) {
+    /**
+     * The related axis of the tick.
+     * @name Highcharts.Tick#axis
+     * @type {Highcharts.Axis}
+     */
     this.axis = axis;
+    /**
+     * The logical position of the tick on the axis.
+     * @name Highcharts.Tick#pos
+     * @type {number}
+     */
     this.pos = pos;
+    /**
+     * The tick type, which can be `"minor"`, or an empty string.
+     * @name Highcharts.Tick#type
+     * @type {string}
+     */
     this.type = type || '';
     this.isNew = true;
     this.isNewLabel = true;
     this.parameters = parameters || {};
-    // Usually undefined, numeric for grid axes
+    /**
+     * The mark offset of the tick on the axis. Usually `undefined`, numeric for
+     * grid axes.
+     * @name Highcharts.Tick#tickmarkOffset
+     * @type {number|undefined}
+     */
     this.tickmarkOffset = this.parameters.tickmarkOffset;
     this.options = this.parameters.options;
     if (!type && !noLabel) {
@@ -86,7 +106,17 @@ H.Tick.prototype = {
             dateTimeLabelFormat = dateTimeLabelFormats.main;
         }
         // set properties for access in render method
+        /**
+         * True, if the tick is the first one on the axis.
+         * @name Highcharts.Tick#isFirst
+         * @type {boolean|undefined}
+         */
         tick.isFirst = isFirst;
+        /**
+         * True, if the tick is the last one on the axis.
+         * @name Highcharts.Tick#isLast
+         * @type {boolean|undefined}
+         */
         tick.isLast = isLast;
         // Get the string
         tick.formatCtx = {
@@ -121,6 +151,11 @@ H.Tick.prototype = {
         }
         // first call
         if (!defined(label)) {
+            /**
+             * The rendered label of the tick.
+             * @name Highcharts.Tick#label
+             * @type {Highcharts.SVGElement|undefined}
+             */
             tick.label = label =
                 defined(str) && labelOptions.enabled ?
                     chart.renderer
@@ -234,15 +269,26 @@ H.Tick.prototype = {
         }
     },
     /**
-     * Get the x and y position for ticks and labels
+     * Gets the x and y positions for ticks.
      *
      * @private
      * @function Highcharts.Tick#getPosition
+     *
      * @param {boolean} horiz
+     * Whether the tick is on an horizontal axis or not.
+     *
      * @param {number} tickPos
+     * Position of the tick.
+     *
      * @param {number} tickmarkOffset
+     * Tickmark offset for all ticks.
+     *
      * @param {boolean} [old]
+     * Whether the axis has changed or not.
+     *
      * @return {Highcharts.PositionObject}
+     * The tick position.
+     *
      * @fires Highcharts.Tick#event:afterGetPosition
      */
     getPosition: function (horiz, tickPos, tickmarkOffset, old) {
@@ -364,6 +410,11 @@ H.Tick.prototype = {
             if (old) {
                 opacity = 0;
             }
+            /**
+             * The rendered grid line of the tick.
+             * @name Highcharts.Tick#gridLine
+             * @type {Highcharts.SVGElement|undefined}
+             */
             tick.gridLine = gridLine = renderer.path()
                 .attr(attribs)
                 .addClass('highcharts-' + (type ? type + '-' : '') + 'grid-line')
@@ -405,6 +456,11 @@ H.Tick.prototype = {
             }
             // First time, create it
             if (isNewMark) {
+                /**
+                 * The rendered mark of the tick.
+                 * @name Highcharts.Tick#mark
+                 * @type {Highcharts.SVGElement|undefined}
+                 */
                 tick.mark = mark = renderer.path()
                     .addClass('highcharts-' + (type ? type + '-' : '') + 'tick')
                     .add(axis.axisGroup);

--- a/ts/parts/Axis.ts
+++ b/ts/parts/Axis.ts
@@ -3852,7 +3852,6 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
          *
          * @see {@link Highcharts.Tick}
          *
-         * @private
          * @name Highcharts.Axis#ticks
          * @type {Highcharts.Dictionary<Highcharts.Tick>}
          */
@@ -3863,7 +3862,6 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
          *
          * @see {@link Highcharts.Tick}
          *
-         * @private
          * @name Highcharts.Axis#minorTicks
          * @type {Highcharts.Dictionary<Highcharts.Tick>}
          */
@@ -4004,9 +4002,14 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
      * The default label formatter. The context is a special config object for
      * the label. In apps, use the
      * [labels.formatter](https://api.highcharts.com/highcharts/xAxis.labels.formatter)
-     * instead except when a modification is needed.
-     * @private
+     * instead, except when a modification is needed.
+     *
+     * @function Highcharts.Axis#defaultLabelFormatter
+     *
+     * @this Highcharts.AxisLabelsFormatterContextObject
+     *
      * @return {string}
+     * The formatted label content.
      */
     defaultLabelFormatter: function (
         this: Highcharts.AxisLabelsFormatterContextObject
@@ -4021,7 +4024,7 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             numSymMagnitude = (lang as any).numericSymbolMagnitude || 1000,
             i = numericSymbols && numericSymbols.length,
             multi,
-            ret,
+            ret: (string|undefined),
             formatOption = (axis.options.labels as any).format,
 
             // make sure the same symbol is added for all labels on a linear
@@ -4034,7 +4037,7 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             ret = format(formatOption, this, time);
 
         } else if (categories) {
-            ret = value;
+            ret = value as any;
 
         } else if (dateTimeLabelFormat) { // datetime axis
             ret = time.dateFormat(dateTimeLabelFormat, value);
@@ -4070,7 +4073,7 @@ H.extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             }
         }
 
-        return ret as any;
+        return ret;
     },
 
     /**

--- a/ts/parts/Tick.ts
+++ b/ts/parts/Tick.ts
@@ -139,23 +139,23 @@ var correctFloat = H.correctFloat,
 /**
  * The Tick class.
  *
- * @private
  * @class
  * @name Highcharts.Tick
  *
  * @param {Highcharts.Axis} axis
+ * The axis of the tick.
  *
  * @param {number} pos
- *        The position of the tick on the axis.
+ * The position of the tick on the axis.
  *
  * @param {string} [type]
- *        The type of tick.
+ * The type of tick.
  *
  * @param {boolean} [noLabel=false]
- *        Wether to disable the label or not. Defaults to false.
+ * Wether to disable the label or not. Defaults to false.
  *
- * @param {Highcharts.TickParametersObject} [parameters]
- *        Optional parameters for the tick.
+ * @param {object} [parameters]
+ * Optional parameters for the tick.
  */
 H.Tick = function (
     this: Highcharts.Tick,
@@ -165,13 +165,33 @@ H.Tick = function (
     noLabel?: boolean,
     parameters?: Highcharts.TickParametersObject
 ): any {
+    /**
+     * The related axis of the tick.
+     * @name Highcharts.Tick#axis
+     * @type {Highcharts.Axis}
+     */
     this.axis = axis;
+    /**
+     * The logical position of the tick on the axis.
+     * @name Highcharts.Tick#pos
+     * @type {number}
+     */
     this.pos = pos;
+    /**
+     * The tick type, which can be `"minor"`, or an empty string.
+     * @name Highcharts.Tick#type
+     * @type {string}
+     */
     this.type = type || '';
     this.isNew = true;
     this.isNewLabel = true;
     this.parameters = parameters || {};
-    // Usually undefined, numeric for grid axes
+    /**
+     * The mark offset of the tick on the axis. Usually `undefined`, numeric for
+     * grid axes.
+     * @name Highcharts.Tick#tickmarkOffset
+     * @type {number|undefined}
+     */
     this.tickmarkOffset = this.parameters.tickmarkOffset;
 
     this.options = this.parameters.options;
@@ -234,7 +254,17 @@ H.Tick.prototype = {
         }
 
         // set properties for access in render method
+        /**
+         * True, if the tick is the first one on the axis.
+         * @name Highcharts.Tick#isFirst
+         * @type {boolean|undefined}
+         */
         tick.isFirst = isFirst;
+        /**
+         * True, if the tick is the last one on the axis.
+         * @name Highcharts.Tick#isLast
+         * @type {boolean|undefined}
+         */
         tick.isLast = isLast;
 
         // Get the string
@@ -278,6 +308,11 @@ H.Tick.prototype = {
         // first call
         if (!defined(label)) {
 
+            /**
+             * The rendered label of the tick.
+             * @name Highcharts.Tick#label
+             * @type {Highcharts.SVGElement|undefined}
+             */
             tick.label = label =
                 defined(str) && labelOptions.enabled ?
                     chart.renderer
@@ -459,15 +494,26 @@ H.Tick.prototype = {
     },
 
     /**
-     * Get the x and y position for ticks and labels
+     * Gets the x and y positions for ticks.
      *
      * @private
      * @function Highcharts.Tick#getPosition
+     *
      * @param {boolean} horiz
+     * Whether the tick is on an horizontal axis or not.
+     *
      * @param {number} tickPos
+     * Position of the tick.
+     *
      * @param {number} tickmarkOffset
+     * Tickmark offset for all ticks.
+     *
      * @param {boolean} [old]
+     * Whether the axis has changed or not.
+     *
      * @return {Highcharts.PositionObject}
+     * The tick position.
+     *
      * @fires Highcharts.Tick#event:afterGetPosition
      */
     getPosition: function (
@@ -686,6 +732,11 @@ H.Tick.prototype = {
             if (old) {
                 opacity = 0;
             }
+            /**
+             * The rendered grid line of the tick.
+             * @name Highcharts.Tick#gridLine
+             * @type {Highcharts.SVGElement|undefined}
+             */
             tick.gridLine = gridLine = renderer.path()
                 .attr(attribs)
                 .addClass(
@@ -757,6 +808,11 @@ H.Tick.prototype = {
 
             // First time, create it
             if (isNewMark) {
+                /**
+                 * The rendered mark of the tick.
+                 * @name Highcharts.Tick#mark
+                 * @type {Highcharts.SVGElement|undefined}
+                 */
                 tick.mark = mark = renderer.path()
                     .addClass('highcharts-' + (type ? type + '-' : '') + 'tick')
                     .add(axis.axisGroup);

--- a/ts/parts/Tick.ts
+++ b/ts/parts/Tick.ts
@@ -146,13 +146,13 @@ var correctFloat = H.correctFloat,
  * The axis of the tick.
  *
  * @param {number} pos
- * The position of the tick on the axis.
+ * The position of the tick on the axis in terms of axis values.
  *
  * @param {string} [type]
- * The type of tick.
+ * The type of tick, either 'minor' or an empty string
  *
  * @param {boolean} [noLabel=false]
- * Wether to disable the label or not. Defaults to false.
+ * Whether to disable the label or not. Defaults to false.
  *
  * @param {object} [parameters]
  * Optional parameters for the tick.
@@ -172,7 +172,7 @@ H.Tick = function (
      */
     this.axis = axis;
     /**
-     * The logical position of the tick on the axis.
+     * The logical position of the tick on the axis in terms of axis values.
      * @name Highcharts.Tick#pos
      * @type {number}
      */
@@ -255,14 +255,16 @@ H.Tick.prototype = {
 
         // set properties for access in render method
         /**
-         * True, if the tick is the first one on the axis.
+         * True if the tick is the first one on the axis.
          * @name Highcharts.Tick#isFirst
+         * @readonly
          * @type {boolean|undefined}
          */
         tick.isFirst = isFirst;
         /**
-         * True, if the tick is the last one on the axis.
+         * True if the tick is the last one on the axis.
          * @name Highcharts.Tick#isLast
+         * @readonly
          * @type {boolean|undefined}
          */
         tick.isLast = isLast;
@@ -494,7 +496,7 @@ H.Tick.prototype = {
     },
 
     /**
-     * Gets the x and y positions for ticks.
+     * Gets the x and y positions for ticks in terms of pixels.
      *
      * @private
      * @function Highcharts.Tick#getPosition

